client/src/components/application/ResponsiveDrawer/list.js

```diff
@@ -0,0 +1,124 @@
+    const PathName = pathname.split('/')[1];
+    let indexElement;
+    // eslint-disable-next-line no-return-assign
+    array.map((element, index) =>
+      element === PathName ? (indexElement = index) : ''
+    );
+
+    if (array.includes(PathName)) {
+      // eslint-disable-next-line no-plusplus
+      for (let i = 0; i <= array.length - 1; i++) {
+        if (i === indexElement) {
+          document
+            .getElementById(`${i}`)
+            .classList.add(`${classes.buttonEffect}`);
+          // eslint-disable-next-line no-continue
+          continue;
+        }
+        document
+          .getElementById(`${i}`)
+          .classList.remove(`${classes.buttonEffect}`);
+      }
+    }
+  }, [pathname]);
+
+  return (
+    <div>
+      <List>
+        {content.map((e, index) => (
+          <ListItem
+            index={index}
+            id={index}
+            button
+            key={e.text}
+            className={classes.list}
+            onClick={(event) => {
+              history.push(`/${e.url}`);
+              const {
+                target,
+                target: { parentElement, tagName },
+              } = event;
+              const elementIndex =
+                target.getAttribute('index') ||
+                parentElement.getAttribute('index') ||
+                parentElement.parentElement.getAttribute('index');
+
+              const newArr = [...Finished];
+              newArr[elementIndex] = !Finished[elementIndex];
+              setFinished(newArr);
+
+              // eslint-disable-next-line no-plusplus
+              for (let i = 0; i <= content.length - 1; i++) {
+                if (i === elementIndex) {
+                  // eslint-disable-next-line no-continue
+                  continue;
+                }
+                document
+                  .getElementById(`${i}`)
+                  .classList.remove(`${classes.buttonEffect}`);
+              }
+              if (tagName === 'DIV' && target.getAttribute('role'))
+                target.classList.add(`${classes.buttonEffect}`);
+              else if (
+                tagName === 'SPAN' &&
+                !target.getAttribute('aria-hidden')
+              )
+                parentElement.parentElement.classList.add(
+                  `${classes.buttonEffect}`
+                );
+              else parentElement.classList.add(`${classes.buttonEffect}`);
+            }}
+          >
+            {Finished[index] ? (
+              <Icon className="far fa-check-circle" />
+            ) : (
+              <Icon className="far fa-times-circle" />
+            )}
+            <ListItemText
+              primary={e.text}
+              classes={{ primary: classes.listText }}
+            />
+          </ListItem>
+        ))}
+      </List>
+    </div>
+  );
+};
+
+DrawerList.propTypes = {
+  content: PropTypes.node,
+  isFinished: PropTypes.node,
+};
+DrawerList.defaultProps = {
+  content: [],
+  isFinished: [false, false],
```

Comment:
is it right to duplicate false two times ?? 
I think you can only add false for once here and declare isFinished as bool not node prototype ?